src/controllers/sagesseController.ts

```diff
@@ -1,234 +1,270 @@
+        details = {'personalWork':personalWork ,
+          'CM':CM ,
+          'TD':TD ,
+          'CMTD':CMTD ,
+          'TP':TP ,
+          'Projet':projet ,
+          'Terrain':terrain ,
+          'description':description ,
+          'context':context ,
+          'content':content,
+          'modules':[]
+        };
+        res.type('application/json');
+        res.status(200);
+      }
+    } catch(error){
+      console.log('Problem while scrapping HTML (element not found)');
+      details = {'error':'can\'t get elements'};
+      res.status(500);
     }
 
-
-
-
-  })
+    res.send(details); //Send the response
+  });
 };
 
 
-export const getYearDetails = async (yearId: string, year: string, res: any) => {
-  Request.get("https://sagesse.polytech.umontpellier.fr/syllabus/"+yearId+"/"+year, (error: any, response: any, body: any) => {
+export const getYearDetails = async (yearId: string, year: string, res: Response) => {
+  Request.get('https://sagesse.polytech.umontpellier.fr/syllabus/'+yearId+'/'+year, (error: any, response: ResponseReq, body: any) => {
     if(error) {
-        return console.dir(error);
+      return console.dir(error);
     }
-    const dom = new JSDOM(body)
-
-    let descripTableNode = dom.window.document.getElementsByClassName("syllabus-elp")[0].querySelector("dl")
-    let hourTableNode = descripTableNode.querySelector("dd").querySelector("dl").querySelectorAll("dd")[4].querySelector("tbody").querySelector("tr")
-
-    let personalWork = descripTableNode.querySelector("dd").querySelector("dl").querySelectorAll("dd")[3].innerHTML
-    let CM = hourTableNode.querySelectorAll("td")[0].textContent
-    let CMTD = hourTableNode.querySelectorAll("td")[1].textContent
-    let TD = hourTableNode.querySelectorAll("td")[2].textContent
-    let TP = hourTableNode.querySelectorAll("td")[3].textContent
-    let terrain = hourTableNode.querySelectorAll("td")[4].textContent
-    let projet = hourTableNode.querySelectorAll("td")[5].textContent
-
-    let description = descripTableNode.children[5].textContent.trim()
-    let context = descripTableNode.children[7].textContent.trim()
-    let content = descripTableNode.children[9].textContent.trim()
-
-    let semesters = Array.from(dom.window.document.getElementsByClassName("listeDescendants")[0].children[1].children).map(subject => {
-      let subjectSplit = (subject as Element)!.querySelector("p")!.textContent!.split(":")
-      let id = subjectSplit[0].trim()
-      let title = subjectSplit[1].trim()
-      return {"id":id, "title":title}
-    })
-
-    let details = {"personalWork":personalWork ,
-      "CM":CM ,
-      "TD":TD ,
-      "CMTD":CMTD ,
-      "TP":TP ,
-      "Projet":projet ,
-      "Terrain":terrain ,
-      "description":description ,
-      "context":context ,
-      "content":content ,
-      "semesters":semesters }
+    const dom = new JSDOM(body);
+
+    let details : any = {};
```

Comment:
here also

(In most of cases we should use a real type and not any ) ;)

Reply:
I add the "object" type for details